### PR TITLE
chore: migrating circleci to github workflows

### PR DESCRIPTION
**Pull request by akeshk**
## Description
This PR migrates the package to use GitHub workflows instead of circleCI for CI.

### DevQA Steps
- Run the pull-request workflow with on: pull_request instead of pull_request_target to verify it passes on the branch.

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+env:
+  CI: 1
+  ARTIFACT_DIR: ./artifacts
+  FORCE_COLOR: 1
+
+jobs:
+  build:
+    name: Install node modules and build
+    runs-on: ubuntu-18.04
```
Comment (noahnu) [inline .github/workflows/pull-request.yml:13]: (nb) you could update this to `ubuntu-20.04`

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      - name: Semantic release
+        run: yarn run semantic-release --dry-run
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: build-artifacts
+          path: ${{ env.ARTIFACT_DIR }}
+  tests:
+    name: Lint and Unit Tests
+    runs-on: ubuntu-18.04
```
Comment (noahnu) [inline .github/workflows/pull-request.yml:47]: (nb) could update this to

File: .github/workflows/pull-request.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+          key: ${{ env.NODE_VERSION }}-${{ env.YARN_VERSION }}-${{ hashFiles('yarn.lock') }}
+      - name: Installl yvm
+        run: node ./scripts/install.js
+      - name: Install dependencies
+        if: steps.node-modules-cache.outputs.cache-hit != 'true'
+        run: make node_modules
+      - name: Build
+        run: make build-production
+      - name: Semantic release
+        run: yarn run semantic-release --dry-run
```
Comment (noahnu) [inline .github/workflows/pull-request.yml:39]: If we're using the github action, I'd just swap this out for the action in dry run mode.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+jobs:
+  release:
+    name: Deploy package
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: cycjimmy/semantic-release-action@v2
+        with:
+          branch: master
+          dry_run: false
```
Comment (noahnu) [inline .github/workflows/release.yml:16]: You can test this by enabling the dry_run.

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+          extra_plugins: |
+            @semantic-release/changelog
+            @semantic-release/git
```
Comment (noahnu) [inline .github/workflows/release.yml:19]: Use the `.releaserc.js` file as a reference. You're missing some plugins. If you have difficulty using the correct plugins, you can continue to just use `yarn semantic-release`

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+          path: ./node_modules
+          key: ${{ env.NODE_VERSION }}-${{ env.YARN_VERSION }}-${{ hashFiles('yarn.lock') }}
+      - run: yarn install --frozen-lockfile
+      - run: yarn build
+      - uses: actions/checkout@v2
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@3.7.1
+        with:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+          BRANCH: gh-pages
```
Comment (noahnu) [inline .github/workflows/release.yml:44]: You can test this by deploying to a custom test branch. E.g. `gh-pages-test` (just make the branch first)

**Review by noahnu — CHANGES_REQUESTED** [review]
Missing semantic release plugins

File: .github/workflows/release.yml (hunk 5 above)
Comment (noahnu) [reply .github/workflows/release.yml:19]: If you do get it to work with github actions, remove the releaserc file if it's no longer in use

File: .github/workflows/pull-request.yml [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+          node-version: ${{ env.NODE_VERSION }}
+      - name: Restore build artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: build-artifacts
+          path: ${{ env.ARTIFACT_DIR }}
+      - name: Install dependencies
+        if: steps.node-modules-cache.outputs.cache-hit != 'true'
+        run: make node_modules
+      - name: Installl yvm
```
Comment (noahnu) [inline .github/workflows/pull-request.yml:104]:
```suggestion
      - name: Install yvm
```

**Review by noahnu — APPROVED** [review]
I'll merge

**Comment by noahnu** [comment]
@all-contributors please add @akeshk for infrastructure, tools, and code

**Comment by noahnu** [comment]
@all-contributors please add @noahnu for infrastructure

File: .github/workflows/pull-request.yml [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+        id: node-modules-cache
+        uses: actions/cache@v2
+        with:
+          path: ./node_modules
+          key: ${{ env.NODE_VERSION }}-${{ env.YARN_VERSION }}-${{ hashFiles('yarn.lock') }}
+      - name: Install yvm
+        run: node ./scripts/install.js
+      - name: Install dependencies
+        if: steps.node-modules-cache.outputs.cache-hit != 'true'
+        run: make node_modules
```
Comment (iamogbz) [inline .github/workflows/pull-request.yml:35]: Shouldn't this run regardless of a cache hit, that was happening in the previous circle config

**Pull request merged**